src/request_flags.rs

```diff
@@ -0,0 +1,322 @@
+use pyo3::prelude::*;
+use pyo3::types::PyBytes;
+
+use crate::{MA_MODE_INC, SET_MODE_SET};
+
+#[pyclass]
+pub struct RequestFlags {
+    #[pyo3(get, set)]
+    no_reply: bool,
+    #[pyo3(get, set)]
+    return_client_flag: bool,
+    #[pyo3(get, set)]
+    return_cas_token: bool,
+    #[pyo3(get, set)]
+    return_value: bool,
+    #[pyo3(get, set)]
+    return_ttl: bool,
+    #[pyo3(get, set)]
+    return_size: bool,
+    #[pyo3(get, set)]
+    return_last_access: bool,
+    #[pyo3(get, set)]
+    return_fetched: bool,
+    #[pyo3(get, set)]
+    return_key: bool,
+    #[pyo3(get, set)]
+    no_update_lru: bool,
+    #[pyo3(get, set)]
+    mark_stale: bool,
+    #[pyo3(get, set)]
+    cache_ttl: Option<u32>,
+    #[pyo3(get, set)]
+    recache_ttl: Option<u32>,
+    #[pyo3(get, set)]
+    vivify_on_miss_ttl: Option<u32>,
+    #[pyo3(get, set)]
+    client_flag: Option<u32>,
+    #[pyo3(get, set)]
+    ma_initial_value: Option<u64>,
+    #[pyo3(get, set)]
+    ma_delta_value: Option<u64>,
+    #[pyo3(get, set)]
+    cas_token: Option<u32>,
+    #[pyo3(get, set)]
+    opaque: Option<Vec<u8>>,
+    #[pyo3(get, set)]
+    mode: Option<u8>,
+}
+
+impl RequestFlags {
+    pub fn push_bytes(&self, buf: &mut Vec<u8>) {
+        if self.no_reply {
+            buf.push(b' ');
+            buf.push(b'q');
+        }
+        if self.return_client_flag {
+            buf.push(b' ');
+            buf.push(b'f');
+        }
+        if self.return_cas_token {
+            buf.push(b' ');
+            buf.push(b'c');
+        }
+        if self.return_value {
+            buf.push(b' ');
+            buf.push(b'v');
+        }
+        if self.return_ttl {
+            buf.push(b' ');
+            buf.push(b't');
+        }
+        if self.return_size {
+            buf.push(b' ');
+            buf.push(b's');
+        }
+        if self.return_last_access {
+            buf.push(b' ');
+            buf.push(b'l');
+        }
+        if self.return_fetched {
+            buf.push(b' ');
+            buf.push(b'h');
+        }
+        if self.return_key {
+            buf.push(b' ');
+            buf.push(b'k');
+        }
+        if self.no_update_lru {
+            buf.push(b' ');
+            buf.push(b'u');
+        }
+        if self.mark_stale {
+            buf.push(b' ');
+            buf.push(b'I');
+        }
+        if let Some(v) = self.cache_ttl {
```

Comment:
Where is `v` defined?

Reply:
This is equivalent to `if v := self.cache_ttl` in python... `v` becomes the value of `cache_ttl` if it's `Some`. The if will not enter if the `Option` is `None`